Frontend/app/problem/display/displayProblem.html

```diff
@@ -0,0 +1,74 @@
+<div class="row">
+  <div class="col-xs-12">
+    <h2>
+      Solving a Problem
+    </h2>
+    <p class="lead" *ngIf="problem">
+      Equation I'm trying to solve: <em>{{problem}}</em>
+    </p>
+  </div>
+</div>
+<!-- show if the problem has loaded -->
+<div *ngIf="!isLoading">
+  <div *ngFor="let step of problemTree; let goodStepIndex = index;">
+    <div class="row">
+      <div class="col-xs-12">
+        <div class="panel panel-default" *ngIf="goodStepIndex <= activeStepCount && !step.badStepShown">
+          <div class="panel-heading">
+            <h3 class="panel-title">
+              {{step.correctStep}}
+            </h3>
+          </div>
+          <div class="panel-body" *ngIf="goodStepIndex == activeStepCount">
+            <button class="btn btn-primary btn-sm" (click)="showNextStep()" *ngIf="goodStepIndex==activeStepCount">
+              Conintue to Next Step
+            </button>
+            <button class="btn btn-danger btn-sm" (click)="incorrectGoodStep()" *ngIf="goodStepIndex==activeStepCount && goodStepIndex != 0">
+              This Step is Incorrect
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+    <!-- loop throguh the bad steps for this step if they exist -->
```

Comment:
For the algebra.py comment, the error says that we couldn't generate the path which is true if the problem is already solved, there wouldn't be a solution path just the solution. Do you want me to change the first part of the error message from Problem Solve Error to something else?